### DISCUSSION TITLE
Icon branch not showing on my mac terminal 🤔

### DISCUSSION THREAD

**Discussion by fiqryq** [starship/starship#3056] [Q&A] [answered]
why is my terminal showing '?' instead of the branch icon? Does anyone have a solution?

<img width="630" alt="Screen Shot 2021-09-12 at 09 49 50" src="https://user-images.githubusercontent.com/25787603/132969821-3c6eed2a-a7a2-4061-b3b9-99c8d7674ef7.png">

**Comment by chipbuster** [comment, marked as answer]
This is happening because you don't have a powerline-patched font, or because your terminal isn't using that.

I'd recommend installing a nerd font (e.g. [via brew-cask](https://github.com/ryanoasis/nerd-fonts#option-4-homebrew-fonts)) and then setting your Terminal to use it as the font.

**Reply by fiqryq** [reply]
hoo its working  thanks mate @chipbuster 

<img width="757" alt="Screen Shot 2021-09-14 at 11 00 24" src="https://user-images.githubusercontent.com/25787603/133192923-c8c812ec-8e7f-4b7c-9991-90bcfff2e3dd.png">

**Reply by chipbuster** [reply]
Woohoo!

It looks like your terminal doesn't correctly understand glyph widths, since your (I think it's NPM?) symbol is getting a little squished with the version.

Unfortunately, the long-term fix is for the terminal program to fix how it understands unicode, so you'd either need to switch terminal applications (e.g. to iTerm2) or wait for Apple to update Terminal.App.

However, if you're okay with sussing out things bit-by-bit, you can modify your config to include two spaces after the symbol, instead of just one. This will cause the terminal to correctly space out the glyphs (the downside is that you have to do this for every symbol that doesn't display correctly in your terminal, and I have no idea of knowing which ones this will be).

To fix Node.js, you would add the following to your config:

```
[nodejs]
symbol = "  "  # Two spaces after symbol to correct for Terminal.App rendering bugs
```
